Downstream patch.

From: Stefan Liebler <stli@linux.ibm.com>
Date: Tue, 6 May 2025 14:49:22 +0200
Subject: [PATCH 2/2] S390: Add z17 to legacy hwcap/platform mechanism

    As glibc 2.39 is still generating ld.so.cache entries for libraries
    in /usr/lib64/z16/libtest.so, do the same for z17 to have the same behavior.
    
    See upstream glibc commits:
    commit b78ff5a25dc8ba9d8c6df10bb0a533254bdd193f
    'elf: Remove legacy hwcaps support from ldconfig'
    https://sourceware.org/git/?p=glibc.git;a=commit;h=b78ff5a25dc8ba9d8c6df10bb0a533254bdd193f
    
    commit c5aa5fd40adc81c4f0b18e01f329aeaf86518c7b
    'elf: Remove loading legacy hwcaps/platform entries in dynamic loader'
    https://sourceware.org/git/?p=glibc.git;a=commit;h=c5aa5fd40adc81c4f0b18e01f329aeaf86518c7b

diff --git a/sysdeps/s390/dl-procinfo-s390.c b/sysdeps/s390/dl-procinfo-s390.c
index 2821e40cb141249c..95cd3c38396c7f26 100644
--- a/sysdeps/s390/dl-procinfo-s390.c
+++ b/sysdeps/s390/dl-procinfo-s390.c
@@ -28,5 +28,5 @@ const char _dl_s390_cap_flags[_DL_HWCAP_COUNT][9] =
 const char _dl_s390_platforms[_DL_PLATFORMS_COUNT][7] =
   {
     "g5", "z900", "z990", "z9-109", "z10", "z196", "zEC12", "z13", "z14", "z15",
-    "z16"
+    "z16", "z17"
   };
diff --git a/sysdeps/s390/dl-procinfo.h b/sysdeps/s390/dl-procinfo.h
index 38e77f7ad9858701..c77557cf57d94fef 100644
--- a/sysdeps/s390/dl-procinfo.h
+++ b/sysdeps/s390/dl-procinfo.h
@@ -23,7 +23,7 @@
 #define _DL_HWCAP_COUNT 23
 extern const char _dl_s390_cap_flags[_DL_HWCAP_COUNT][9] attribute_hidden;
 
-#define _DL_PLATFORMS_COUNT	11
+#define _DL_PLATFORMS_COUNT	12
 extern const char _dl_s390_platforms[_DL_PLATFORMS_COUNT][7] attribute_hidden;
 
 /* The kernel provides up to 32 capability bits with elf_hwcap.  */
